posts/_markdown/quarc.md

```diff
@@ -0,0 +1,24 @@
+---
+title: QUARC
+date: January 23, 2023
+author: Christoph Kirsch and Stefanie Muroya Lei
+---
+
+Bounded Model Checking (BMC) is a well-established technique used to ensure the correctness of critical software. Given a number *n* representing the number of state transitions, it builds the Finite State Machine (FSM) of a program and searches for the input(s) that make a set *S* of machine states (e.g. division by zero, non-zero exit codes) reachable within *n* state transitions.
+
+
+BMC algorithms use heuristics and Satisfiability Modulo Theory (SMT) solvers that do not scale in general since all solve an SAT formula whose solutions grow exponentially in the number of paths of length *n* due to branching and loop conditions that depend on input values.
+ 
+
+That is the reason why we created QUARC: a backend included in the [Unicorn toolchain](https://github.com/cksystemsgroup/unicorn). Unicorn supports RISC-V, and it takes binaries of arbitrary programs to produce an inner representation of an FSM that it then optimizes through SAT and Satisfiability-Modulo Theory Solvers.
+
+QUARC takes as input a number *n* and the FSM of a program to produce an oracle that represents *n* state transitions of these programs. This oracle outputs true whenever some state of *S* happens in *n* or fewer state transitions.
```

Comment:
It would be great to clearly draw the line between the previous Unicorn project and the QUARC project funded here. Also in terms of language, functionalities it implements and community it addresses.